core/src/main/java/io/questdb/std/Files.java

```diff
@@ -173,12 +173,39 @@ public static long floorPageSize(long size) {
 
     public static native int fsync(int fd);
 
+    public static long getDirectoryContentSize(Path path) {
+        long pathUtf8Ptr = path.address();
+        long pFind = findFirst(pathUtf8Ptr);
+        if (pFind > 0L) {
+            int len = path.length();
+            try {
+                long totalSize = 0L;
+                do {
+                    long nameUtf8Ptr = findName(pFind);
+                    Chars.utf8DecodeZ(nameUtf8Ptr, path.trimTo(len).slash$());
+                    path.$();
+                    if (findType(pFind) == Files.DT_FILE) {
+                        totalSize += length0(pathUtf8Ptr);
+                    } else if (notDots(nameUtf8Ptr)) {
+                        totalSize += getDirectoryContentSize(path);
+                    }
+                }
+                while (findNext(pFind) > 0);
+                return totalSize;
+            } finally {
+                findClose(pFind);
+                path.trimTo(len).$();
+            }
+        }
+        return 0L;
+    }
+
     public static long getDiskSize(LPSZ path) {
         if (path != null) {
             return getDiskSize(path.address());
         }
         // current directory
-        return getDiskSize(0);
+        return 0L;
```

Comment:
no, that is a bug fix. If path is not null (the most common branch) then we enter the `getDiskSize` function passing its off-heap address. Otherwise size is zero.